magenta-lib/src/main/scala/magenta/input/resolver/DeploymentGraphBuilder.scala

```diff
@@ -0,0 +1,46 @@
+package magenta.input.resolver
+
+import magenta.graph.{Edge, EndNode, Graph, MidNode, Node, StartNode}
+import magenta.input.Deployment
+
+object DeploymentGraphBuilder {
```

Comment:
It's another approach. You wind up building edges multiple times with that approach as there are potentially many paths through the graph. You have to start at the StartNode and build onwards recursing down. think you might have to pass through an accumulator of visited nodes as well.

I just had a bash and it gets complicated very quickly. Let's chat IRL.